File: server/engine/player.go
package engine

type player struct {
	name string
	color color
	timer string
	capturedPieces []piece 
}

func createPlayer(n string, c color) player {
	return player{
		name: n,
		color: c,
		timer: "3:00",
		capturedPieces: []piece{},
	}
}

File: server/src/engine/player.ts
import { Color, ChessPiece } from "../types/general.enums";

export default class Player {
    private username: string;
    private color: Color;
    private timer: string;
    private capturedPieces: ChessPiece[];

    constructor(name: string, color: Color) {
        this.username = name;
        this.color = color;
        this.timer = "3:00";
        this.capturedPieces = []
    }

    json(): string {
        return JSON.stringify({username: this.username, color: Color[this.color]});
    }
}